Update Issue and PR templates (#726)
GregTechCEu/GregTech: .github/pull_request_template.md

**What:**
_This section describes what this PR is about. It should be a clear and concise description concerning why is this PR needed and why it should be accepted.

> _Link to explanation from issue this PR is solving can be used instead of copying same text here._

Reviewer: ```suggestion
_Linking an issue can be used alternatively to writing a description._
```

**Implementation Details:**
_Any implementations in this PR that should be carefully looked over, or that could/should have alternate solutions recommended._

**Outcome:**
_This section is used mostly for generating the changelog and there should be a short described what this PR added/fixed/changed/removed._
_For correct linking of issues please use any of Closes/Fixes/Resolves keywords. Example: When PR is fixing a bug use "Fixes: #number-of-bug"_

**Additional info:**

> _This section is for screenshots to demonstrate any GUI, rendering or recipe changes. Or any additional info that reviewers should be aware of._

Reviewer: ```suggestion
_This section is for screenshots to demonstrate any GUI, rendering, recipe changes, or any additional info that reviewers should be aware of._
```
